### DISCUSSION TITLE
monai.transforms.SaveImage saves image with the axis swapped - help needed

### DISCUSSION THREAD

**Discussion by soumbane** [Project-MONAI/MONAI#4681] [Q&A] [answered]
Hi,
I have an image of shape: (2, 128, 128, 64), i.e., 2 channels (0 for CT and 1 for MRI), each modality with 128 x 128 shaped images with 64 slices.
I load it by using:
loader = monai.transforms.LoadImage(dtype=np.float32)
img, _ = loader(train_data_dicts[0]["image"]) # img shape: (2, 128, 128, 64)

I want to save (in "*.nii.gz" format) the channel 1 (MR) separately to view on 3D Slicer software. So, I save it as:
out_dir = os.path.join(data_dir, "Data_3D")
saver = monai.transforms.SaveImage(output_dir=out_dir, output_postfix='saved', output_ext='.nii.gz', 
                  output_dtype=np.float32, resample=False, scale=None, dtype=None, squeeze_end_dims=False, 
                  data_root_dir='', separate_folder=False, print_log=False)
saver(img[1], meta_data={'filename_or_obj':'MR_img1'})
print(np.shape(img[1])) # (128, 128, 64)

The file is saved as: "MR_img1_saved.nii.gz" in the "Data_3D" folder.

When I try to load this image again using:
loader = monai.transforms.LoadImage(dtype=np.float32)
img, _ = loader('.../Data_3D/MR_img1_saved.nii.gz')
print(np.shape(img)) # (128, 64, 1, 128)

The loaded img shape is (128,64,1,128) and this is the reason I cannot view the file "MR_img1_saved.nii.gz" on 3D slicer.

Please let me know the problem here and how do I fix it?

Thanks.

**Comment by JohnMasoner** [comment, marked as answer]
Hey @soumbane ,

Maybe the docs could help you. [SaveImage docs](https://docs.monai.io/en/stable/transforms.html?highlight=SaveImage#saveimage)

The Parameters of __call__ of the SaveImage Class is `img `and `meta_data ` where the requirements of `img ` are as following:

```
img (Union[Tensor, ndarray]) – target data content that save into file. The image should be channel-first, shape: [C,H,W,[D]].
```

shape is [C,H,W,[D]], but yours is [H, W, D].

Hope this is useful to you.

**Reply by Nic-Ma** [reply]
And the `squeeze_end_dim` parameter may also be useful to remove dim:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/io/array.py#L299

Thanks.

**Reply by soumbane** [reply]
Thank you @Nic-Ma and @JohnMasoner for your help. It works with the squeeze_end_dim=True after I add a dimension to (128,128,64) to make it (1,128,128,64).
